torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -444,6 +465,10 @@ IndexCompute IndexCompute::updateIndexCompute(
       updated_index_map[new_id] = index_map_.at(prev_id);
     }
 
+    if (!prev_id->isBroadcast() && new_id->isBroadcast()) {
```

Comment:
Can you give some comments here? `updated_extent_map` is set at three different paths with `extent_map_.at(prev_id)` or `getExtent(prev_id)`. Would be nice why each path would be taken and why it would use `extent_map_` or `getExtent`.

Reply:
I think this should actually be an `if else` in the `if` below. This is primarily just not setting an entry in the extent map unless it's needed. That way we know if extent has diverged from what's in the `IterDomain`. `getExtent` is just a convenience wrapper that will return a map entry if there is one, otherwise will return the extent of the `IterDomain` provided.

Reply:
The one I actually don't get below is the is reduction section, uncertain if or why that'd be needed. Another thing to keep in mind, is I think we only update the index compute once per indexing now, instead of through out a large tv stack so we may be able to simplify this code somewhat.

Reply:
Thanks. I think this part of the change is also related. The propagation through the `isReduction()` logic may not kick in now, and that seems to explain why an axis can be missing in `extent_map` .

Reply:
I'll revisit this later again.